crates/redis_interface/src/commands.rs

```diff
@@ -174,6 +174,31 @@ impl super::RedisConnectionPool {
             .change_context(errors::RedisError::SetExFailed)
     }
 
+    #[instrument(level = "DEBUG", skip(self))]
+    pub async fn set_key_if_not_exists_with_expiry<I, O>(
+        &self,
+        key: &str,
+        value: I,
+        seconds: i64,
+    ) -> CustomResult<Option<O>, errors::RedisError>
+    where
+        I: TryInto<RedisValue> + Debug + Send + Sync,
+        I::Error: Into<fred::error::RedisError> + Send + Sync,
+        O: FromRedis,
+    {
+        self.pool
+            .set(
+                key,
+                value,
+                Some(Expiration::EX(seconds)),
+                Some(SetOptions::NX),
+                true,
+            )
+            .await
+            .into_report()
+            .change_context(errors::RedisError::SetFailed)
+    }
+
```

Comment:
We already have a function `set_key_if_not_exist` which sets default expiry you can just pass `Option<i64>` in that function and set that expiry if you pass `Some(expiry_seconds)` otherwise it sets the default expiry. I don't think we need a separate function for it.

Reply:
cc: @SanchithHegde @vspecky

Reply:
Will do this